lcls_tools/data_analysis/archiver.py

```diff
@@ -18,82 +20,97 @@
 
         response = requests.post(url=url, data={"pv": ",".join(pvList)},
                                  timeout=TIMEOUT)
 
-        results = {}
+        values = {}
+        times = [timeRequested]
+        for pv in pvList:
+            values[pv] = []
 
         try:
             jsonData = json.loads(response.text)
-            for key, val in jsonData.items():
-                results[key] = val[u'val']
+            for pv, data in jsonData.items():
+                values[pv].append(data[u'val'])
 
         except ValueError:
             print("JSON error with {PVS} at {TIME}".format(PVS=pvList,
                                                            TIME=timeRequested))
-        return results
+        return ArchiverData(timeStamps=times, values=values)
 
     def getDataWithTimeInterval(self, pvList, startTime, endTime, timeDelta):
-        # type: (List[str], datetime, datetime, timedelta) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
+        # type: (List[str], datetime, datetime, timedelta) -> ArchiverData
         currTime = startTime
-        results = {}
+        times = []
+        values = {}
         for pv in pvList:
-            results[pv] = {"times": [], "values": []}
+            values[pv] = []
 
         while currTime < endTime:
             result = self.getDataAtTime(pvList, currTime)
-            for key, val in result.items():
-                results[key]["times"].append(currTime)
-                results[key]["values"].append(val)
+            times.append(currTime)
+            for key, val in result.values.items():
+                values[key].append(val.pop())
             currTime += timeDelta
 
-        return results
+        return ArchiverData(timeStamps=times, values=values)
 
     # returns timestamps in UTC
     def getValuesOverTimeRange(self, pvList, startTime, endTime, timeDelta=None):
-        # type: (List[str], datetime, datetime, timedelta) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
+        # type: (List[str], datetime, datetime, timedelta) -> ArchiverData
 
         if timeDelta:
             return self.getDataWithTimeInterval(pvList, startTime, endTime, timeDelta)
         else:
             url = self.url_formatter.format(SUFFIX=RANGE_RESULT_SUFFIX)
 
-            results = {}
+            times = {}
+            values = {}
 
             # TODO figure out how to send all PVs at once
             for pv in pvList:
+                times[pv] = []
+                values[pv] = []
 
                 response = requests.get(url=url, timeout=TIMEOUT,
-                                        params={"pv": pv,
+                                        params={"pv"  : pv,
                                                 "from": startTime.isoformat() + "-07:00",
-                                                "to": endTime.isoformat() + "-07:00"})
+                                                "to"  : endTime.isoformat() + "-07:00"})
 
```

Comment:
Maybe add a comment here about the -7 offset for people who don't know why we need that.